activerecord/test/cases/timestamp_test.rb

```diff
@@ -253,6 +253,56 @@ def change_name
     assert_not_equal @previously_updated_at, @developer.updated_at
   end
 
+  def test_saving_an_unchanged_record_with_a_mutating_before_update_callback_updates_its_timestamp
+    klass = Class.new(ActiveRecord::Base) do
+      self.table_name = "developers"
+
+      include Developer::TimestampAliases
+
+      before_update :change_name
+
+      private
+        def change_name
+          return if new_record?
+
+          self.name = "Jack Bauer"
+        end
+    end
+
+    @developer = klass.create!
+    @previously_updated_at = @developer.updated_at
+    @previous_name = @developer.name
+
+    travel(1.second) do
+      @developer.save!
+    end
+
+    assert_not_equal @previous_name, @developer.name
+    assert_not_equal @previously_updated_at, @developer.updated_at
```

Comment:
Failing assertion:
<img width="1333" alt="Screenshot 2023-03-25 at 5 56 32 pm" src="https://user-images.githubusercontent.com/54629302/227705097-c5592fb2-72c4-4bc1-97d6-8784a6c85a7f.png">

Reply:
```suggestion
    assert_not_equal @previously_updated_at, @developer.updated_at
    assert_not_equal @previously_updated_at, @developer.reload.updated_at
```

This ensures that the correct timestamp is persisted, too.

Reply:
(Should do the same on the other test.)

Reply:
Actioned!
https://github.com/rails/rails/compare/ba69cd2d2130cf6d42bd201da9b67ebd756ec4a4..f2d622adeb4847361c64bdb6d9b0ce30b9b106fb